File: testcorpus/go/sample2.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

type wordCount struct {
	word  string
	count int
}

func countWords(text string) map[string]int {
	counts := map[string]int{}
	for _, word := range strings.Fields(strings.ToLower(text)) {
		word = strings.Trim(word, ".,!?;:")
		if word != "" {
			counts[word]++
		}
	}
	return counts
}

func topWords(counts map[string]int, limit int) []wordCount {
	ranked := make([]wordCount, 0, len(counts))
	for word, count := range counts {
		ranked = append(ranked, wordCount{word: word, count: count})
	}
	sort.Slice(ranked, func(i, j int) bool {
		if ranked[i].count == ranked[j].count {
			return ranked[i].word < ranked[j].word
		}
		return ranked[i].count > ranked[j].count
	})
	if limit < len(ranked) {
		return ranked[:limit]
	}
	return ranked
}

func main() {
	text := "the quick brown fox jumps over the lazy dog the fox"
	for _, entry := range topWords(countWords(text), 3) {
		fmt.Printf("%s: %d\n", entry.word, entry.count)
	}
}

File: testcorpus/c/sample2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

typedef struct {
    char *data;
    size_t len;
    size_t cap;
} buffer_t;

static int buffer_init(buffer_t *buf, size_t cap) {
    buf->data = malloc(cap);
    if (buf->data == NULL) {
        return -1;
    }
    buf->len = 0;
    buf->cap = cap;
    return 0;
}

static int buffer_push(buffer_t *buf, const char *text) {
    size_t need = strlen(text) + 1;
    if (buf->len + need > buf->cap) {
        size_t next = buf->cap * 2;
        char *grown = realloc(buf->data, next);
        if (grown == NULL) {
            return -1;
        }
        buf->data = grown;
        buf->cap = next;
    }
    memcpy(buf->data + buf->len, text, need);
    buf->len += need - 1;
    return 0;
}

int count_words(const char *text) {
    int words = 0;
    int in_word = 0;
    for (const char *p = text; *p != '\0'; p++) {
        if (isspace((unsigned char)*p)) {
            in_word = 0;
        } else if (!in_word) {
            in_word = 1;
            words++;
        }
    }
    return words;
}

int main(void) {
    buffer_t buf;
    if (buffer_init(&buf, 64) != 0) {
        return 1;
    }
    buffer_push(&buf, "the quick brown fox");
    buffer_push(&buf, " jumps over the lazy dog");
    printf("%s\n", buf.data);
    printf("words: %d\n", count_words(buf.data));
    free(buf.data);
    return 0;
}
